com.heroicgameslauncher.yml

```diff
@@ -0,0 +1,145 @@
+  - --share=network
+  - --socket=x11
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --allow=multiarch
+  - --allow=devel
+  - --device=all
+  - --talk-name=org.freedesktop.Notifications
+  - --persist=.wine
+  - --filesystem=home
+  - --filesystem=xdg-data/Steam:ro
+  - --filesystem=~/.var/app/com.valvesoftware.Steam:ro
+  - --require-version=1.1.2
+  - --env=XDG_CONFIG_DIRS=/etc/xdg:/usr/lib/x86_64-linux-gnu/GL:/usr/lib/i386-linux-gnu/GL
+  - --env=GST_PLUGIN_SYSTEM_PATH=/app/lib/gstreamer-1.0:/app/lib32/gstreamer-1.0:/usr/lib/extensions/gstreamer-1.0:/usr/lib/x86_64-linux-gnu/gstreamer-1.0:/usr/lib/i386-linux-gnu/gstreamer-1.0
+
+add-extensions:
+  org.freedesktop.Platform.Compat.i386:
+    directory: lib/i386-linux-gnu
+    version: '21.08'
+
+  org.freedesktop.Platform.Compat.i386.Debug:
+    directory: lib/debug/lib/i386-linux-gnu
+    version: '21.08'
+    no-autodownload: true
+
+  org.freedesktop.Platform.GL32:
+    directory: lib/i386-linux-gnu/GL
+    version: '1.4'
+    versions: 21.08;1.4
+    subdirectories: true
+    no-autodownload: true
+    autodelete: false
+    add-ld-path: lib
+    merge-dirs: vulkan/icd.d;glvnd/egl_vendor.d;OpenCL/vendors;lib/dri;lib/d3d;vulkan/explicit_layer.d;vulkan/implicit_layer.d
+    download-if: active-gl-driver
+    enable-if: active-gl-driver
+
+modules:
+
+  # --- Tools ---
+  - name: unappimage
+    buildsystem: simple
+    build-commands:
+      - make -C squashfs-tools -j ${FLATPAK_BUILDER_N_JOBS} install INSTALL_DIR=/app/bin
+    sources:
+      - type: git
+        url: https://github.com/refi64/unappimage
+        commit: d7f86f2a0d7ec3a69211125207d5f127386b849a
+
+  - name: vulkan-tools
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DGLSLANG_INSTALL_DIR=/app
+      - -DVULKAN_HEADERS_INSTALL_DIR=/app
+      - -DCMAKE_BUILD_TYPE=Release
+    sources:
+      - type: archive
+        url: https://github.com/KhronosGroup/Vulkan-Tools/archive/refs/tags/v1.2.186.tar.gz
+        sha256: 8ba0e21e8de0b70d31985d251d11150efee71266875262798784e9d1ec9b717b
+
+  - name: gamemode
+    buildsystem: meson
+    config-opts: &gamemode_opts
+      - -Dwith-systemd=false
+      - -Dwith-daemon=false
+      - -Dwith-examples=false
+      - -Dwith-util=false
+      - -Dwith-sd-bus-provider=no-daemon
+    sources: &gamemode_sources
+      - type: archive
+        url: https://github.com/FeralInteractive/gamemode/releases/download/1.6.1/gamemode-1.6.1.tar.xz
+        sha256: 10c2a3f142eae472f5a09e42616e38c666c05b25ca3e61e562a543bb3fda66c5
+
+  - name: gamemode-32bit
+    build-options:
+      arch:
+        x86_64: &compat_i386_opts
+    buildsystem: meson
+    config-opts: *gamemode_opts
+    sources: *gamemode_sources
+
+  - name: gamemoderun
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 data/gamemoderun -t /app/bin
+    sources: *gamemode_sources
+
+  # --- Heroic ---
+  - name: heroic
+    buildsystem: simple
+    build-commands:
+      - unappimage Heroic-2.2.2.AppImage
```

Comment:
What's the rationale for pulling the AppImage version here, instead of e.g. pulling the source and building it?

Reply:
Simplicity and effectiveness since pulling the whole source code, requiring NodeJS runtime, Python, and several other dependencies. Making the build completely bloated.
The AppImage runs pretty well, its compiled and optimized already.
So, unless there is something bad about using it, I believe we could stick to it :)

Reply:
I can't make a compelling case neither in favour or against this choice because I don't know AppImage well enough to catch subtleties. However, I have the following concerns about it:

 * How does the AppImage work with regards to libc? Does it bundle it? Is mixing the unpacked AppImage with Flatpak platform libraries safe?
 * Which distro / platform is this AppImage built against?
 * If AppImage statically links dependencies, are you sure they wouldn't clash with dynamically linked libraries from the Flatpak runtime?

(To be abundantly clear, I am not playing devil's advocate; I'm sure you have good reasons to have opted for this approach, I just want to understand these reasons better.)

Reply:
Yes, no worries.
I am not an AppImage expert as well. :)
But this is an Electron App built with Electron Builder.
In summary, all Linux packages that we have today, also the Windows and the Mac ones, are the same thing, just a folder with all files build without external dependencies. so if I extract the AppImage and the Deb or the Pacman file, the content will be pretty similar, respecting only the package type differences.

Reply:
> How does the AppImage work with regards to libc? Does it bundle it? Is mixing the unpacked AppImage with Flatpak platform libraries safe?

It does nothing with libc and does not contain it. It may be broken if the developer builds against a newer libc but generally they develop against old libc to be safe. None